tests/unit/test_column_noise.py

```diff
@@ -60,9 +68,37 @@ def test_generate_missing_data(string_series, default_configuration):
     assert (string_series[not_noised_idx] == noised_data[not_noised_idx]).all()
 
 
-@pytest.mark.skip(reason="TODO")
-def test_incorrect_selection():
-    pass
+def test_incorrect_selection(categorical_series, default_configuration):
+    config = default_configuration["decennial_census"]["state"]["incorrect_selection"]
+    noised_data = generate_incorrect_selections(
+        categorical_series, config, RANDOMNESS0, "test_incorrect_select"
+    )
+    noised_data_same_seed = generate_incorrect_selections(
+        categorical_series, config, RANDOMNESS0, "test_incorrect_select"
+    )
+    noised_data_different_seed = generate_incorrect_selections(
+        categorical_series, config, RANDOMNESS1, "test_incorrect_select"
+    )
+
+    # Confirm same randomness stream provides same results
+    assert (noised_data == noised_data_same_seed).all()
+
+    # Confirm different streams provide different results
+    assert (noised_data != noised_data_different_seed).any()
+
+    # Check for expected noise level
+    expected_noise = config["row_noise_level"]
+    # todo: Update when generate_incorrect_selection uses exclusive resampling
+    # Get real expected noise to account for possibility of noising with original value
+    # Here we have a a possibility of choosing any of the 50 states for our categorical series fixture
+    expected_noise = expected_noise * (1 - 1 / 50)
+    actual_noise = (noised_data != categorical_series).sum() / len(noised_data)
+    assert np.isclose(expected_noise, actual_noise, rtol=0.02)
+
+    # Check that un-noised values are unchanged
+    not_noised_idx = noised_data.index[noised_data == categorical_series]
+    assert (categorical_series[not_noised_idx] == noised_data[not_noised_idx]).all()
```

Comment:
Isn't this assertion always true by how you defined `not_noised_idx`? (I think I did the same thing on a test I previously wrote)

I'm actually not sure if there is a useful test to check that un-noised values are unchanged. 

I suppose the assertian below that all noised data is notna is nice, but I don't think there are guarantees that incomming data are notna so this will break in that case.

Reply:
Agreed this assert is not actually testing anything.